src/ui/component-repository/components/item/install-controls/InstallButton.js

```diff
@@ -18,6 +18,28 @@ const jobProgressStatuses = [
   ECR_COMPONENT_UNINSTALLATION_STATUS_IN_PROGRESS,
 ];
 
+const compareSemanticVersions = (a, b) => {
+  // 1. Split the strings into their parts.
+  const a1 = a.split('.');
+  const b1 = b.split('.');
+  // 2. Contingency in case there's a 4th or 5th version
+  const len = Math.min(a1.length, b1.length);
+  // 3. Look through each version number and compare.
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < len; i++) {
+    const a2 = +a1[i] || 0;
+    const b2 = +b1[i] || 0;
+
+    if (a2 !== b2) {
+      return a2 > b2 ? 1 : -1;
+    }
+  }
+
+  // 4. We hit this if the all checked versions so far are equal
+  //
+  return b1.length - a1.length;
+};
+
```

Comment:
I think you should move this function inside the helper directory (inside helper/versions.js I think can be a good place). You should also add the tests for this function.